ext/openssl/xp_ssl.c

```diff
@@ -456,13 +485,17 @@ static bool php_openssl_matches_san_list(X509 *peer, const char *subject_name) /
 				if (strcasecmp(subject_name, (const char*)ipbuffer) == 0) {
 					sk_GENERAL_NAME_pop_free(alt_names, GENERAL_NAME_free);
 
+					return 1;
+				}
+			} else if (san->d.ip->length == 16 && strlen(ipv6_expanded) >= 15) { /* shortest expanded IPv6 address is 0:0:0:0:0:0:0:0 */
```

Comment:
it would be probably better to add some flag that it was expanded rather than always check strlen...

Reply:
done